Dockerfile

```
@@ -0,0 +1,26 @@
+FROM python:3.11-slim as base
+RUN apt-get -y update; apt-get -y install curl
+RUN curl -sSL https://install.python-poetry.org | python3 -
+ENV PATH="/root/.local/bin:$PATH"
+RUN apt-get update && apt-get install -y firefox-esr curl
+COPY . .
+RUN poetry install --no-root --without dev
+COPY ./.vscode ./.vscode
+COPY ./fixtures ./fixtures
+COPY ./todo_app ./todo_app
```

Is there any reason for these later `COPY` statements if everything is copied initially anyway? Would this work better as a targeted `COPY` step first, with all other changes coming after?
(One extra COPY added to solve an important file this misses, though arguably that should happen in the test stage?)
```suggestion
COPY pyproject.toml poetry.lock ./
RUN poetry install --no-root --without dev
COPY ./.vscode ./.vscode
COPY ./fixtures ./fixtures
COPY ./todo_app ./todo_app
COPY .env.test .env.test
```

---

Hm yep this was definitely my aim, I think I must have got half way through doing this and then forgotten to finish the job 😅